[lz4] Add "tools" feature

anarazel opened the pull request:
I would like to use the lz4 binary in tests when building with lz4 support. Unfortunately the vcpkg port of lz4 does not build the lz4 binary. This PR adds a feature, defaulting to off, to enable tools to be built.

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [x] SHA512s are updated for each updated download
- [x] The "supports" clause reflects platforms that may be fixed by this new version
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

anarazel committed:
[lz4] Add "tools" feature

Adela0814 commented on ports/lz4/portfile.cmake:
```diff
@@ -8,8 +8,16 @@ vcpkg_from_github(
 
 file(COPY "${CMAKE_CURRENT_LIST_DIR}/CMakeLists.txt" DESTINATION "${SOURCE_PATH}")
 
+if("tools" IN_LIST FEATURES)
+   set(LZ4_BUILD_TOOLS 1)
+else()
+   set(LZ4_BUILD_TOOLS 0)
+endif()
```
vcpkg_check_features(OUT_FEATURE_OPTIONS FEATURE_OPTIONS
    FEATURES
        "tools" LZ4_BUILD_TOOLS
)

anarazel replied:
Done.

anarazel commented:
I updated the PR to address @Adela0814's feedback. I also added a call to `vcpkg_copy_tools(TOOL_NAMES lz4 AUTO_CLEAN)` as apparently tools are not supposed to be in bin/. To support non-static linking I added the lib sources to lz4cli's sources, as lz4cli uses internal APIs.

Adela0814 commented on ports/lz4/portfile.cmake:
```diff
@@ -8,8 +8,15 @@ vcpkg_from_github(
+
 vcpkg_cmake_configure(
     SOURCE_PATH "${SOURCE_PATH}"
+    OPTIONS
+        -DLZ4_BUILD_TOOLS=${LZ4_BUILD_TOOLS}
```
```suggestion
    OPTIONS
        ${FEATURE_OPTIONS}
```

dg0yt replied:
I would also add 
~~~
    OPTIONS_DEBUG
        -DLZ4_BUILD_TOOLS=OFF
~~~
(The debug variant of the tools will be thrown away anyways.)

dg0yt commented on ports/lz4/CMakeLists.txt:
```diff
@@ -46,6 +64,24 @@ install(EXPORT lz4Config
+    set_target_properties(lz4cli PROPERTIES OUTPUT_NAME lz4)
+    target_include_directories(lz4cli PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_LIST_DIR}/lib> $<INSTALL_INTERFACE:include>)
+    if(BUILD_STATIC_LIBS)
+        target_link_libraries(lz4cli lz4)
```
Given that the executable uses `LZ4_SOURCES` *and* `LZ4CLI_SOURCES`, 
- why link to lz4 lib here?
- why add `LZ4CLI_SOURCES` to `LZ4_SOURCES` in L36?

Adela0814 commented:
Pinging @anarazel  Can you address the review suggestions? Thanks!

Adela0814 commented:
Closing this PR since it seems that no progress is being made. Please ping us to reopen if work is still being done.
